Automattic/wp-calypso
# Update transferred domain item statuses

### Changes proposed in this Pull Request

This PR updates the transferred domain item statuses in the site domains management page. This is part of the domain transfer flow redesign detailed in pcYYhz-hK-p2.

#### Screenshots

These are the new transferred domain statuses:

- Transfer pending - the limit date to begin the transfer is 25 days after the transfer subscription is bought
<img width="1180" alt="Screen Shot 2021-09-23 at 17 58 57" src="https://user-images.githubusercontent.com/5324818/134584066-14f74d18-942a-4120-9401-2ee8e055d4a2.png">

- Transfer in progress
<img width="1188" alt="Screen Shot 2021-09-23 at 18 01 38" src="https://user-images.githubusercontent.com/5324818/134584172-3ff4a63d-0db5-4c32-aefd-a6344f8938da.png">

- Transfer in progress with estimated finish time
<img width="1180" alt="Screen Shot 2021-09-23 at 18 02 50" src="https://user-images.githubusercontent.com/5324818/134584206-7a2705de-0239-410e-b562-f86b592e5164.png">

- Transfer failed
<img width="1182" alt="Screen Shot 2021-09-23 at 18 00 39" src="https://user-images.githubusercontent.com/5324818/134584112-8116bded-e668-4308-b9f2-cee76f2d611f.png">

- Transfer completed successfully but domain does not resolve to WPCOM yet
<img width="1184" alt="Screen Shot 2021-09-23 at 18 05 09" src="https://user-images.githubusercontent.com/5324818/134584307-ac62a220-cb46-4dfb-9a80-36a20abf5885.png">

### Testing instructions

- This should be tested with D67325-code applied to your sandbox.
- Open the live Calypso link, check the site domains management page (Upgrades > Domains) and try to find transferred domains in the following states:
  - Transfer not started (when a user chooses to use a transferred domain during onboarding)
  - Transfer in progress
  - Transfer failed
  - Transfer completed successfully and the domain does not point to WPCOM's name servers yet
- Alternatively, run this branch locally and hack `resolve-domain-status.js` to force the desired domain statuses on your domains.
- Ensure all these situations have the correct status messages and layout as described in the screenshots and pcYYhz-hK-p2.
